errdefs/errdefs.go

```diff
@@ -0,0 +1,28 @@
+package errdefs
+
+import "errors"
+
+type internalErr struct {
+	error
+}
+
+func (internalErr) System() {}
+
+func (err internalErr) Unwrap() error {
+	return err.error
+}
+
+type system interface {
+	System()
+}
+
+var _ system = internalErr{}
+
+func Internal(err error) error {
```

Comment:
The separate function is mostly to avoid direct `grpcerrors` dependency in packages that don't interact with grpc. Otherwise `WrapCode()` is the generic version where any code can be passed directly.